patoolib/__init__.py

```diff
@@ -558,7 +622,23 @@ def get_archive_cmdlist_func (program, command, format):
         raise util.PatoolError(msg)
     # get archive handler function (eg. patoolib.programs.star.extract_tar)
     try:
-        return getattr(module, '%s_%s' % (command, format))
+        archive_cmdlist_func = getattr(module, '%s_%s' % (command, format))
+        def check_for_password_before_cmdlist_func_call(*args, **kwargs):
+            """ If password is None, or not set, run command as usual.
+            If password is set, but can't be accepted raise appropriate
+            message.
+            """
+            util.log_info("... cmdlist_func = %s %s" % (archive_cmdlist_func, ''))
+            util.log_info("... kwargs=%s args=%s" % (kwargs, args))
+            if 'password' in kwargs and kwargs['password'] is None:
```

Comment:
minor: could be just 

```suggestion
            if kwargs.get('password') is None:
```
to avoid duplication, and make code shorter/even more readable

Reply:
It will be True when there is no 'password' in kwargs, so pop will raise exception.
Of course, I can use suppress, but it will be one more line, so I'm not sure it'll be better.

Reply:
ah, right, I just misread the logic - my bad.  I think then it is ok as is.

FWIW, it could be `.pop("password", None)` to avoid an exception if not there.